docs/learn/documentation/versioned/azure/eventhubs.md

```diff
@@ -0,0 +1,105 @@
+
+# define required properties for your streams
+systems.eh-system.streams.input0.eventhubs.namespace=YOUR-STREAM-NAMESPACE
+systems.eh-system.streams.input0.eventhubs.entitypath=YOUR-ENTITY-NAME
+systems.eh-system.streams.input0.eventhubs.sas.keyname=YOUR-SAS-KEY-NAME
+systems.eh-system.streams.input0.eventhubs.sas.token=YOUR-SAS-KEY-TOKEN
+
+systems.eh-system.streams.output0.eventhubs.namespace=YOUR-STREAM-NAMESPACE
+systems.eh-system.streams.output0.eventhubs.entitypath=YOUR-ENTITY-NAME
+systems.eh-system.streams.output0.eventhubs.sas.keyname=YOUR-SAS-KEY-NAME
+systems.eh-system.streams.output0.eventhubs.sas.token=YOUR-SAS-KEY-TOKEN
+```
+
+The tuple required to access the Eventhubs entity per stream must be provided, namely the fields `YOUR-STREAM-NAMESPACE`, `YOUR-ENTITY-NAME`, `YOUR-SAS-KEY-NAME`, `YOUR-SAS-KEY-TOKEN`.
+
+### Producing to EventHubs:
+
+Similarly, you can also configure your Samza job to write to EventHubs.  
+```
+OutgoingMessageEnvelope envelope = new OutgoingMessageEnvelope(new SystemStream("eh-system", "output0"), key, message);
+collector.send(envelope);
+```
+The first parameter in the `OutgoingMessageEnvelope` is the name of the system and the output stream whose properties are specified in config. The `key` field in the envelope is used as the key in the `EventData` while the`message` field is used as its body.
+
+The envelopes sent will be converted to an `EventData` that is accepted by EventHub. The envelope `message` will be the body of the `EventData` and the envelope "key" and "produce time stamp" will be included in the properties map as `key` and `produce-timestamp` respectively. Currently there are no ways to pass other metadata in the properties field of `EventData`.
+
+### Advanced configuration:
+
+##### Producer partitioning: 
+
+The `partition.method` property determines how outgoing messages are partitioned. Valid values for this config are `EVENT_HUB_HASHING`, `PARTITION_KEY_AS_PARTITION` or `ROUND_ROBIN`. 
+
+The `EVENT_HUB_HASHING` is the default setting for the producer. This method employs the hashing mechanism in EventHubs to determine, based on the key of the message, which partition the message should go. Using this method still ensures that all the events with the same key are sent to the same partition in the event hub. If this option is chosen, the partition key used for the hash should be a string. If the partition key is not set, the message key is used instead.
+
+The `PARTITION_KEY_AS_PARTITION` method will use the integer key specified by the partition key or key of the message to a specific partition on Eventhub. If the integer key is greater than the number of partitions in the destination Eventhub, a modulo operation will be performed to determine the resulting partition. ie. if there are 6 partitions and the key is 9, the message will end up in partition 3. Similarly to `EVENT_HUB_HASHING`, if the partition key is not set the message key is used instead.
```

Comment:
PARTITION_KEY_AS_PARTITION: In this method, each message is sent to the partition specified by its partition key. This requires the partition key to be an integer. If the key is greater than the number of partitions, a modulo operation will be performed on the key. Similar to EVENT_HUB_HASHING, the key in the message is used if the partition key is not specified.